spec/warden/proxy_spec.rb

```diff
@@ -409,25 +409,59 @@
       end
       setup_rack(app).call(@env)
     end
+
+    it "should set renew on rack.session.options" do
+      app = lambda do |env|
+        env['warden'].authenticate(:pass)
+        valid_response
+      end
+
+      @env[Warden::Proxy::ENV_SESSION_OPTIONS] = {}
+
+      setup_rack(app).call(@env)
+
+      expect(@env[Warden::Proxy::ENV_SESSION_OPTIONS]).to include(renew: true)
+      expect(@env[Warden::Proxy::ENV_SESSION_OPTIONS]).to_not be_frozen
+    end
+
+    it "should not modify attempt to modify a frozen rack.session.options" do
+      app = lambda do |env|
+        env['warden'].authenticate(:pass)
+        valid_response
+      end
+
+      original_options = {}.freeze
```

Comment:
@md5 ah this is where we struggled. We had this working, but then when trying to test that Rack itself did in fact freeze the hash, we were unable to reproduce that behavior.